### DISCUSSION TITLE
(When) can Ditto be upgraded to a newer version while keeping the uploaded Things?

### DISCUSSION THREAD

**Discussion by stfnp** [eclipse-ditto/ditto#1556] [Q&A] [answered]
Hello everyone!

I was wondering what the conditions are (if any) for safely upgrading Ditto to a newer version. Will the database contents still be compatible with a newer version or should breaking changes be expected?

We are runnng the Docker deployment of Ditto with the standard MongoDB container and would like to upgrade from time to time, while ideally keeping the Things that have been uploaded already. Of course they could be reuploaded, but we would like to avoid/prepare for the disruption if possible.

I couldn't find an answer to this in the documentation, but maybe I've just missed it.

**Comment by thjaeckle** [comment, marked as answer]
Hi @stfnp 

Good question and yes, it is not explicitly stated in the docs.

TL;DR: Yes, the entities (things, policies and connections) are always kept when updating Ditto.


Each release notes contain "Migration notes" (if there are any) to upgrade from the former version to the new version of Ditto.
Things/policies and connections are always kept compatible - We ensured this since version 1.0.0 of Ditto.

What changed recently in Ditto 3.0.0 is for example the search-index of things .. the index however can be restored by the information from the "things" and "policies" persistence - and that is what Ditto 3.0.0 does, it automatically indexes all things in the new search index.

We performed all updgrades since Ditto 1.0.0 and never "lost" a thing ;)
